### PR TITLE
Add product brief doc.

### PR DESCRIPTION
**Pull request by chokoswitch**

### CODE REVIEW

**Comment by chokoswitch** [comment]
@skywhale Here is a project brief for Curiostack

File: docs/project-brief.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+- A more thorough design document is needed for probing the technical aspects of Curiostack.
+- Is there a finer balance needed between easy-to-use pre-baked opinions and user configurability?
+- Is the minimum cost of a Google Container Engine cluster too high for experimentation?
+
+### Timeline
+
+##### Milestone 1
+- Single command that creates a GCP project, GKE cluster, sets up Google Container Builder for
+autobuild and deploy
+- An authentication service based on Firebase - LINE login support is nice to have
```
Comment (demongaorui) [inline docs/project-brief.md]: Line don't include users' email in their open api, however for many services they have to obtain at least one verified contact approach(email or phone number normally) to reach their users. While using Line login could not give the service an efficient way to contact the users.

File: docs/project-brief.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+Engine (GKE) cluster, creating databases, and other common cluster operations.
+- Build system support for setting up client/server API-based services with minimal boilerplate.
+- Automatic test and beta deployment of services.
+- Security first. All endpoints will use HTTPs, and security best practices will be baked in without
+the need for user configuration.
+
+### Challenges and open questions
+
+- A more thorough design document is needed for probing the technical aspects of Curiostack.
+- Is there a finer balance needed between easy-to-use pre-baked opinions and user configurability?
```
Comment (demongaorui) [inline docs/project-brief.md]: I think make easy-to-use pre-baked opinions as default and also give users chance to customize if they really need to and know what they are doing is a good and widely used way. Then you could focus on finding the best default config according to the best practices you believe. The advanced users will check these source codes anyway if they are really advanced 😉

File: docs/project-brief.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+- Is the minimum cost of a Google Container Engine cluster too high for experimentation?
```
Comment (demongaorui) [inline docs/project-brief.md]: IMO, yes. Especially, the new users will have a concern whether this edge tech really could help them to build a stable service for their startup career. While maybe if we could using this stack build some awesome on-line applications, then this problem will be resolved.

**Pull request closed without merging**